gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/converter/HiveAvroToOrcConverter.java

```diff
@@ -0,0 +1,168 @@
+import gobblin.converter.DataConversionException;
+import gobblin.converter.SchemaConversionException;
+import gobblin.converter.SingleRecordIterable;
+import gobblin.data.management.conversion.hive.HiveSource;
+import gobblin.data.management.conversion.hive.entities.QueryBasedHiveConversionEntity;
+import gobblin.util.AvroFlattener;
+import gobblin.data.management.conversion.hive.util.HiveAvroORCQueryUtils;
+
+/**
+ * Builds the Hive avro to Orc conversion query. The record type for this converter is {@link QueryBasedHiveConversionEntity}. A {@link QueryBasedHiveConversionEntity}
+ * can be a hive table or a hive partition.
+ */
+@Slf4j
+public class HiveAvroToOrcConverter
+    extends Converter<Schema, Schema, QueryBasedHiveConversionEntity, QueryBasedHiveConversionEntity> {
+
+  // TODO: Remove when topology is enabled
+  private static final String ORC_TABLE_ALTERNATE_LOCATION = "orc.table.alternate.location";
+
+  private static AvroFlattener avroFlattener = new AvroFlattener();
+
+  @Override
+  public Schema convertSchema(Schema inputSchema, WorkUnitState workUnit) throws SchemaConversionException {
+    return inputSchema;
+  }
+
+  /**
+   * Populate the avro to orc conversion query which will be available at
+   * {@link QueryBasedHiveConversionEntity#getConversionQuery()}
+   *
+   */
+  @Override
+  public Iterable<QueryBasedHiveConversionEntity> convertRecord(Schema outputSchema,
+      QueryBasedHiveConversionEntity conversionEntity, WorkUnitState workUnit) throws DataConversionException {
+    Preconditions.checkNotNull(outputSchema);
+    Preconditions.checkNotNull(conversionEntity);
+    Preconditions.checkNotNull(workUnit);
+    Preconditions.checkArgument(conversionEntity.getHiveUnit().getLocation().isPresent());
+    Preconditions.checkArgument(StringUtils.isNotBlank(conversionEntity.getHiveUnit().getTableName()));
+
+    Schema flattenedSchema = avroFlattener.flatten(outputSchema, true);
+
+    // Create flattened table if not exists
+    // ORC Hive tables are named as   : {avro_table_name}_orc
+    // ORC Hive tables use location as: {avro_table_location}_orc
+    // TODO: Add cluster by info (from config)
+    // TODO: Add sort by info (from config)
+    // TODO: Add num of buckets (from config)
+
+    // Avro table name and location
+    String avroTableName = conversionEntity.getHiveUnit().getTableName();
+    String avroTableLocation = conversionEntity.getHiveUnit().getLocation().get();
+
+    // ORC table name and location
+    String orcTableName = avroTableName + "_orc";
+    String orcTableLocation;
+
+    // By default ORC table creates a new directory where Avro data resides with _orc postfix, but this can be
+    // .. overridden by specifying this property
+    String orcTableAlternateLocation = workUnit.getJobState().getProp(ORC_TABLE_ALTERNATE_LOCATION);
+    if (StringUtils.isNotBlank(orcTableAlternateLocation)) {
+      orcTableLocation = orcTableAlternateLocation.endsWith("/") ?
```

Comment:
Can you use ```StringUtils.removeEnd(str, remove)```. I think it takes care of removing the end ```\``` only if it exits. Also for slashes use ```Path.SEPARATOR```